swap/src/bitcoin/wallet.rs

```diff
@@ -1637,6 +1917,116 @@ fn estimate_fee(
     Ok(amount)
 }
 
+mod mempool_client {
+    static HTTP_TIMEOUT: Duration = Duration::from_secs(15);
+    static BASE_URL: &str = "https://mempool.space";
+
+    use super::EstimateFeeRate;
+    use anyhow::{bail, Context, Result};
+    use bitcoin::{FeeRate, Network};
+    use serde::Deserialize;
+    use std::time::Duration;
+
+    /// A client for the mempool.space API.
+    ///
+    /// This client is used to estimate the fee rate for a transaction.
+    pub struct MempoolClient {
+        client: reqwest::Client,
+        base_url: String,
+    }
+
+    #[derive(Deserialize)]
+    struct MempoolFees {
+        #[serde(rename = "fastestFee")]
+        fastest_fee: u64,
+        #[serde(rename = "halfHourFee")]
+        half_hour_fee: u64,
+        #[serde(rename = "hourFee")]
+        hour_fee: u64,
+        #[serde(rename = "minimumFee")]
+        minimum_fee: u64,
+    }
```

Comment:
I think you can just do this 

```suggestion
    #[derive(Deserialize)]
    #[serde(rename_all = "camelCase")]
    struct MempoolFees {
        fastest_fee: u64,
        half_hour_fee: u64,
        hour_fee: u64,
        minimum_fee: u64,
    }
```
https://serde.rs/attr-rename.html